addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -443,11 +444,27 @@ class MessageList extends Component {
             return false;
         }
         const loadMoreRect = loadMore.getBoundingClientRect();
-        const elRect = this.el.getBoundingClientRect();
+        const elRect = this.scrollableParent.getBoundingClientRect();
         const isInvisible = loadMoreRect.top > elRect.bottom || loadMoreRect.bottom < elRect.top;
         return !isInvisible;
     }
 
+    /**
+     * @private
+     * @param {Element}
+     * @returns {Element} closest scrollable node element
+     */
+    getScrollableParent(node) {
```

Comment:
I would rename this to `getClosestScrollableElement` as you correctly used in the docstring because the element can also be this and not a parent. This is similar to the `closest()` Element method.

But see comments below, this can probably be renamed something else.